tests/lib/rules/no-unnecessary-act.test.ts

```diff
@@ -0,0 +1,782 @@
+import { createRuleTester } from '../test-utils';
+import rule, { RULE_NAME } from '../../../lib/rules/no-unnecessary-act';
+
+const ruleTester = createRuleTester();
+
+/**
+ * - AGR stands for Aggressive Reporting
+ * - RTL stands for React Testing Library (@testing-library/react)
+ * - RTU stands for React Test Utils (react-dom/test-utils)
+ */
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    {
+      code: `// case: RTL act wrapping non-RTL calls
+      import { act } from '@testing-library/react'
+
+      test('valid case', async () => {
+        act(() => {
+          stuffThatDoesNotUseRTL();
+        });
+
+        await act(async () => {
+          stuffThatDoesNotUseRTL();
+        });
+
+        act(function() {
+          stuffThatDoesNotUseRTL();
+          const a = foo();
+        });
+
+        act(function() {
+          return stuffThatDoesNotUseRTL();
+        });
+
+        act(() => stuffThatDoesNotUseRTL());
+      });
+      `,
+    },
+    {
+      code: `// case: RTU act wrapping non-RTL
+      import { act } from 'react-dom/test-utils'
+
+      test('valid case', async () => {
+        act(() => {
+          stuffThatDoesNotUseRTL();
+        });
+
+        await act(async () => {
+          stuffThatDoesNotUseRTL();
+        });
+
+        act(function() {
+          stuffThatDoesNotUseRTL();
+        });
+
+        act(function() {
+          return stuffThatDoesNotUseRTL();
+        });
+
+        act(() => stuffThatDoesNotUseRTL());
+      });
+      `,
+    },
+    {
+      settings: {
+        'testing-library/utils-module': 'test-utils',
+      },
+      code: `// case: RTL act wrapping non-RTL - AGR disabled
+      import { act } from '@testing-library/react'
+      import { waitFor } from 'somewhere-else'
+
+      test('valid case', async () => {
+        act(() => {
+          waitFor();
+        });
+
+        await act(async () => {
+          waitFor();
+        });
+
+        act(function() {
+          waitFor();
+        });
+
+        act(function() {
+          return waitFor();
+        });
+
+        act(() => waitFor());
+      });
+      `,
+    },
+    {
+      code: `// case: RTL act wrapping both RTL and non-RTL calls
```

Comment:
If I understand correctly, the exception for allowing both RTL+non-RTL calls is made because of possible use cases that'd be blocked if this is reported as an error. 

On the other hand, I can't think of any such use case and I haven't yet stumbled upon one in my experience, so what I worry about is that users might take this wrong and use this exception to "workaround" the error for regular use cases, when `act()` is indeed unnecessary. This brings the question: if the use cases for RTL+non-RTL calls in `act()` are rare, why not report an error and let users block disable it?

Reply:
[I double-check it in the original issue](https://github.com/testing-library/eslint-plugin-testing-library/issues/259#issuecomment-803413239). Not really strong opinion about it, but we thought it could be useful not to report it.

Reply:
This is an interesting invalid usage! @kentcdodds it seems it would be a good idea to report also mixed RTL and non-RTL calls.

Reply:
Not really sure what to do with this finally. Completely in favor of reporting any `act` with some react-testing-library util call in it, even if mixed with non-testing-library calls. Not sure if there is a really obvious exception to implement or some note to add to the doc about mixed calls.

Reply:
How are you planning to release this rule? Is it gonna be a part of recommended ruleset with minor bump?

Reply:
Minor bump but without including the rule in any existing shareable config, that would be a breaking change for sure.

Reply:
How about reporting mixed RTL and non-RTL calls, assuming that's what users expect, unless someone creates an issue, after which we can reconsider?

Reply:
Sure, we can do that.

Reply:
Wait, but the currently released version **does not** report mixed calls, right?

Reply:
Oh it doesn't. I understood we wanted to leave it unreported and see how it goes. My bad.

Reply:
I'm gonna create an issue for adding a new `isStrict` option to this rule so it reports all `act` with an RTL call inside, no matter if mixed with non-RTL or not.